django/db/models/fields/json.py

```diff
@@ -479,6 +479,27 @@ class KeyTransformGte(KeyTransformNumericLookupMixin, lookups.GreaterThanOrEqual
     pass
 
 
+class KeyTransformIn(lookups.In):
+    def process_rhs(self, compiler, connection):
+        rhs, rhs_params = super().process_rhs(compiler, connection)
+        if connection.vendor == 'oracle':
+            func = []
+            for value in rhs_params:
+                value = json.loads(value)
+                function = 'JSON_QUERY' if isinstance(value, (list, dict)) else 'JSON_VALUE'
+                func.append("%s('%s', '$.value')" % (
+                    function,
+                    json.dumps({'value': value}),
+                ))
+            rhs = rhs % tuple(func)
+            rhs_params = []
+        elif connection.vendor in {'sqlite', 'mysql'}:
+            func = ["JSON_EXTRACT(%s, '$')"] * len(rhs_params)
+            rhs = rhs % tuple(func)
+        return rhs, tuple(rhs_params)
```

Comment:
This is really similar to `process_rhs` in `KeyTransformExact` though, can we reuse that somehow?

Also, this is a rather edge case, but this means one of the elements of the rhs can't be `None` on SQLite (unless we add `OR JSON_TYPE` in the lhs.

Reply:
> This is really similar to process_rhs in KeyTransformExact though, can we reuse that somehow?

I tried but it's really hacky, because part of the logic is in `JSONExact`.

> Also, this is a rather edge case, but this means one of the elements of the rhs can't be None on SQLite (unless we add OR JSON_TYPE in the lhs.

`None` is ignored in `__in` lookup, so that's not an issue.